toolkit/scripts/toolchain/cross/gcc_cross.sh

```diff
@@ -42,52 +47,53 @@ cd gcc-9.1.0
 ln -s ../mpfr-4.0.1 mpfr
 ln -s ../gmp-6.1.2 gmp
 ln -s ../mpc-1.1.0 mpc
-
-# Patch GCC to avoid an error where PATH_MAX isn't defined
-patch -p1 < "${scriptDir}/pathmax.patch"
 cd ..
 
-sudo mkdir -p ${installDir}
+sudo mkdir -p "${installDir}/aarch64-linux-gnu"
 export PATH="${installDir}/bin":$PATH
 
+cd WSL2-Linux-Kernel-linux-msft-5.4.83
+make mrproper
+make ARCH=arm64 headers
+make ARCH=arm64 INSTALL_HDR_PATH=${sysroot} headers_install
+cd ..
+
 mkdir build-binutils
 cd build-binutils
-../binutils-2.32/configure --prefix=${installDir} --target=aarch64-linux-gnu --disable-multilib
+../binutils-2.32/configure --prefix=${installDir} --target=aarch64-linux-gnu --disable-multilib --with-sysroot=${sysroot}
 make -j$(nproc)
 make install
 cd ..
 
-cd WSL2-Linux-Kernel-linux-msft-5.4.83
-make mrproper
-make ARCH=arm64 headers_check
-make ARCH=arm64 INSTALL_HDR_PATH="${installDir}/aarch64-linux-gnu" headers_install
-cd ..
 
 mkdir -p build-gcc
 cd build-gcc
-../gcc-9.1.0/configure --prefix=${installDir} --target=aarch64-linux-gnu --disable-multilib --enable-shared --enable-threads=posix --enable-__cxa_atexit --enable-clocale=gnu --enable-languages=c,c++,fortran --disable-bootstrap --enable-linker-build-id  --enable-plugin --enable-default-pie
+../gcc-9.1.0/configure --prefix=${installDir} --target=aarch64-linux-gnu --disable-multilib --enable-shared --enable-threads=posix --enable-__cxa_atexit --enable-clocale=gnu --enable-languages=c,c++,fortran --disable-bootstrap --enable-linker-build-id  --enable-plugin --enable-default-pie --with-sysroot=${sysroot} --with-native-system-header-dir=/include
 make -j$(nproc) all-gcc
 make install-gcc
 cd ..
 
 mkdir -p build-glibc
 cd build-glibc
-../glibc-2.28/configure --prefix="${installDir}/aarch64-linux-gnu" --build=$MACHTYPE --host=aarch64-linux-gnu --target=aarch64-linux-gnu --with-headers="${installDir}/aarch64-linux-gnu/include" --disable-multilib libc_cv_forced_unwind=yes  --disable-werror
-make install-bootstrap-headers=yes install-headers
-make -j$(nproc) csu/subdir_lib
-install csu/crt1.o csu/crti.o csu/crtn.o "${installDir}/aarch64-linux-gnu/lib"
-aarch64-linux-gnu-gcc -nostdlib -nostartfiles -shared -x c /dev/null -o "${installDir}/aarch64-linux-gnu/lib/libc.so"
-touch "${installDir}/aarch64-linux-gnu/include/gnu/stubs.h"
+../glibc-2.28/configure --prefix=/ --build=$MACHTYPE --host=aarch64-linux-gnu --target=aarch64-linux-gnu --with-sysroot=${sysroot} --with-headers="${sysroot}/include" --disable-multilib libc_cv_forced_unwind=yes  --disable-werror
+make DESTDIR=${sysroot} install-bootstrap-headers=yes install-headers 
+make -j$(nproc) csu/subdir_lib "install_root=${installDir}/aarch64-linux-gnu/sysroot"
```

Comment:
> "install_root=${installDir}/aarch64-linux-gnu/sysroot"

If I'm understanding this correctly, `${sysroot}` is currently set to `${installDir}/aarch64-linux-gnu`. When we build the various tools with `--with-sysroot=`, do the tools create an explicit sysroot directory under `${sysroot}`?

Also should `${sysroot}` be pointing to this sysroot directory instead of `${installDir}/aarch64-linux-gnu`?

Reply:
Thinking on this a bit more - since we are setting `${sysroot}` to be `${installDir}/aarch64-linux-gnu`, it looks like we will end up with a mix of native and target binaries/libs in our `${sysroot}` directory, since the toolchain installs into this same directory. AFAIK sysroot should only contain target collateral so we keep our native and target states separate. Is it possible to cleanly separate out the sysroot from everything native?

Reply:
Good catch. As we discussed yesterday, the target binaries/libs are now separated.